ladybug_geometry/geometry3d/sphere.py

```diff
@@ -0,0 +1,194 @@
+        """Scale a arc by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the sphere should be scaled.
+            origin: A Point2D representing the origin from which to scale.
+                If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        return Sphere(self.center.scale(factor, origin), self.radius * factor)
+
+    def intersect_plane(self, plane):
+        """Get the intersection of a plane with this Sphere object
+
+        Args:
+            plane: A Plane object
+
+        Returns:
+            Arc3D representing a full circle if it exists.
+            None if no intersection exists.
+        """
+        r = self.radius
+        pt_c = self.center
+        pt_o = plane.o
+        v_n = plane.n.normalize()
+
+        # Resulting circle radius. Radius² = r² - [(c-p).n]²
+        v_d = ((pt_c + pt_o) * v_n)
+        d = v_d.magnitude  # d = plane.distance_to_point(self._center)
+        if r < d:  # No intersection if negative
+            return None
+        cut_r = math.sqrt(r ** 2 - d ** 2)
+        if cut_r == 0:  # Tangent intersection ignored - results in a point
+            return None
+
+        # Resulting circle center point. Center_point = p - [(c-p).n]n
+        cut_center = pt_c + v_d * v_n
+        cut_plane = Plane(v_n, cut_center)
+        return Arc3D(cut_plane, cut_r)
+
+    def intersect_line_ray(self, line_ray):
+        """Get the intersection between this Sphere object and a Ray2D/LineSegment2D.
+
+        Args:
+            line_ray: A LineSegment3D or Ray3D that will be extended infinitely
+                for intersection.
+
+        Returns:
+            A list of 2 Point3D objects if a full intersection exists.
+            A list with a single Point3D object if the line is tangent or intersects
+            only once. None if no intersection exists.
+            return
+        """
+        sc = self.center
+        r = self.radius
+        p1 = line_ray.p1
+        p2 = line_ray.p2
+        dp = p2 - p1
+        a = dp.x * dp.x + dp.y * dp.y + dp.z * dp.z
+        b = 2 * (dp.x * (p1.x - sc.x) + dp.y * (p1.y - sc.y) + dp.z * (p1.z - sc.z))
+        c = sc.x * sc.x + sc.y * sc.y + sc.z * sc.z
+        c += p1.x * p1.x + p1.y * p1.y + p1.z * p1.z
+        c -= 2 * (sc.x * p1.x + sc.y * p1.y + sc.z * p1.z)
+        c -= r * r
+        bb4ac = b * b - 4 * a * c
+
+        if abs(a) < math.ldexp(1.0, -53) or bb4ac < 0:
+            return None  # No intersection
+
+        mu1 = (-b + math.sqrt(bb4ac)) / (2 * a)
+        mu2 = (-b - math.sqrt(bb4ac)) / (2 * a)
+        pt1 = p1 + mu1 * (p2 - p1)
+        pt2 = p1 + mu2 * (p2 - p1)
+
+        return [pt1] if mu1 == mu2 else [pt1, pt2]
+
+    def duplicate(self):
+        """Get a copy of this object."""
+        return self.__copy__()
+
+    def to_dict(self):
+        """Get Sphere as a dictionary."""
+        return {'type': 'Sphere', 'center': self.center.to_dict(),
```

Comment:
This should be `self.center.to_array()` if you choose to implement the suggestion I have above. A mistake like this makes it a bit clearer why it will be helpful to have `to_array` methods.